newrelic-agent/src/main/java/com/newrelic/agent/ThreadService.java

```diff
@@ -82,20 +80,7 @@ public ThreadNameNormalizer getThreadNameNormalizer() {
 
     @Override
     public String getThreadName(final BasicThreadInfo threadInfo) {
-        String name = threadIdToName.getIfPresent(threadInfo.getId());
-        if (null == name) {
-            try {
-                return threadIdToName.get(threadInfo.getId(), new Callable<String>() {
-                    @Override
-                    public String call() throws Exception {
-                        return threadInfo.getName();
-                    }
-                });
-            } catch (ExecutionException e) {
-                return threadInfo.getName();
-            }
-        }
-        return name;
+        return threadIdToName.get(threadInfo.getId(), id -> threadInfo.getName());
```

Comment:
Throughout the PR: `ExecutionException` isn't thrown anymore, so I dropped associated `try/catch` blocks where reasonable, taking into account whether the backing code could actually throw an exception and whether the fallback logic "made sense" or seemed to be there just to appease the compiler.